pytorch_lightning/core/datamodule.py

```diff
@@ -76,13 +76,16 @@ def wrapped_fn(*args, **kwargs):
         if fn.__name__ == "setup":
 
             # Get stage either by grabbing from args or checking kwargs.
-            # If not provided, set call status of 'fit' and 'test' to True.
+            # If not provided, set call status of 'fit', 'validation', and 'test' to True.
             # We do this so __attach_datamodule in trainer.py doesn't mistakenly call setup('test') on trainer.test()
             stage = args[1] if len(args) > 1 else kwargs.get("stage", None)
 
             if stage == "fit" or stage is None:
                 obj._has_setup_fit = True
 
+            if stage == "validation" or stage is None:
```

Comment:
I would make all references in datamodule "validate" (instead of "validation) to keep it consistent with fit and test

Reply:
fit and test can be nouns or verbs, however, we are talking about stages which means they should be nouns.
So if I am not mistaken (English is not my first language), using validation is more consistent.
The validation stage vs the validate stage

Reply:
I don't think grammar should be a consideration here since we're only talking about variables in code... and that variable name consistency is more important. Thoughts on this? @justusschock @rohitgr7

Reply:
I think both of you have a point here and once could certainly use both. However, I feel that validation stage is more intuitive and personally I would go with it since it sounds 'more correct' to me, but this is just a personal opinion. Also I think, that this should definitely not be a blocker here.

Reply:
Yes, it is a good point. I was also ambivalent about it while I was writing the code.

There is another occurrence of this issue: the `Trainer.evaluating` attribute, which can be either `test` or `validation`. Here `validation` is the right choice in my opinion, reading it as "currently evaluating over the test/validation set".

It was not so clear cut in the data module: I'd say that 'validation' sounds better for me too, but I would not be opposed to using 'validate' either.

Reply:
Gotcha, it does have a better ring to it :]